snapcraft/plugins/nodejs.py

```diff
@@ -251,6 +264,15 @@ def _build_environment(self):
         else:
             new_path = npm_bin
 
+        # npm has the behavior that, if it detects that SUDO_UID is set,
+        # it will then set the uid of some of its child processes (such as
+        # git when installing a package that specifies a git repository)
+        # to that of the sudoer, ignoring --unsafe-perm entirely.
+        # We have to unset SUDO_UID and SUDO_GID to prevent this.
+
+        env.pop("SUDO_UID", None)
```

Comment:
Did you hit this case by chance?  If so, presumably using destructive-mode?  I'm curious how/when it's set in the environment :)

Reply:
No, this is a known bug/feature in npm. I've made a report about it here, it's fairly easy to replicate: https://npm.community/t/npm-ignores-unsafe-perm-and-user-params-when-running-git-as-part-of-install/10437

Reply:
This is reproduceable @cjp256 and we have a bug for it too https://bugs.launchpad.net/snapcraft/+bug/1816204

Reply:
Thanks!